### ISSUE TITLE
The question for  Cross entropy loss

### ISSUE THREAD

**Issue by storm-zhuo** [luanyunteng/pytorch-be-your-own-teacher#2]
Hi,
Thanks for sharing the code, but i have some questions 

In my understanding, middle_output1, middle_output2, and middle_output3 should be different from output. But in the code, they are the same as model(input)?

And i'm looking forward to the code of L2 loss of intermediate features . Your reply will be higly appreciated!

Best

**Comment by luanyunteng** [comment]
> Hi,
> Thanks for sharing the code, but i have some questions
> 
> In my understanding, middle_output1, middle_output2, and middle_output3 should be different from output. But in the code, they are the same as model(input)?
> 
> And i'm looking forward to the code of L2 loss of intermediate features . Your reply will be higly appreciated!
> 
> Best
hello, I am sorry for my late reply. The middle_outputs are use the same model input, but they are different as the different branches.
